routes/api/v1/boring_query.js

```diff
@@ -0,0 +1,133 @@
+    // look in the BoringQuery table for an existing query matching the request
+    BoringQuery.findOne({
+      where: {
+        query: searchQuery
+      }
+    })
+    .then(query => {
+      //if it is a new query
+      if (!query) {
+        // create url for Edamam using the query from the param
+        const appId = process.env.EDAMAM_ID
+        const appKey = process.env.EDAMAM_KEY
+        url = `https://api.edamam.com/search?q=${searchQuery}&app_id=${appId}&app_key=${appKey}&calories=2000-999999&to=30`
+
+        //fetch new recipes from edamam api
+        getRecipes(url)
+        .then(recipeResponse => {
+          //create BoringQuery, Recipes, and BoringQueryRecipes in database
+          return saveBoringRecipes(recipeResponse, searchQuery);
+        })
+        .then(recipes => {
+          //Send newly retrieved recipes. Sort by highest calorie total and only top 10 results.
+          recipes.sort((a, b) => b.calories - a.calories);
+          recipes = recipes.slice(0, 10);
+          res.status(200).send(JSON.stringify(recipes))
+        })
+        .catch(error => {
+          res.status(404).send({error: error})
+        })
+      //If an existing query is found
+      } else {
+        // find the recipes that are already associated with the existing query
+        Recipe.findAll({
+          include: [{
+            model: BoringQueryRecipe,
+            attributes: [],
+            where: {
+              BoringQueryId: query.id
+            }
+          }],
+          order: [['calories', 'DESC']],
+          limit: 10
+        })
+        .then(recipes => {
+          //Send retrieved recipes.
+          res.status(200).send(JSON.stringify(recipes))
+        })
+        .catch(error => {
+          res.status(404).send({error: error})
+        })
+      }
+    })
+    .catch(error => {
+      res.status(404).send({error: "Bad search query."})
+    })
+  }
+})
+
+function getRecipes(url){
+  return new Promise((resolve, reject) => {
+    fetch(url)
+    .then(response => {
+      if (response.status = 200){
+        resolve(response.json())
+      }else{
+        reject(result.error_message)
+      }
+    })
+  })
+};
+
+function saveBoringRecipes(recipeResponse, searchQuery){
+  return new Promise((resolve, reject) => {
+    BoringQuery.create({
+      query: searchQuery
+    })
+    .then(query => {
+      resolve(parseRecipes(recipeResponse, query));
+    })
+    .catch(error => {
+      reject(error)
+    })
+  })
+};
+
+function parseRecipes(recipeResponse, query) {
+  return Promise.all(recipeResponse.hits.map(recipe => {
+    return saveRecipe(recipe, query);
+  }))
+};
+
+function saveRecipe(recipe, query){
+  return new Promise((resolve, reject) => {
+    Recipe.findOrCreate({
+      where: {name: recipe.recipe.label},
+      defaults: {
+        url: recipe.recipe.url,
+        yield: recipe.recipe.yield,
+        calories: Math.round((recipe.recipe.calories / recipe.recipe.yield)),
```

Comment:
What are your thoughts on saving the recipe calories as `calories per serving` (I'm assuming that's what this is doing) rather than `total calories`?

Reply:
Sure. At least clarifying that in the endpoint section of the readme. Formatting (like certain naming) is ultimately up to the hypothetical front-end. But I agree, it could be changed for clarity.

Reply:
I'll add the user story which documents the endpoint in the README.